commands/operator-sdk/cmd/new.go

```diff
@@ -164,47 +163,114 @@ func doAnsibleScaffold() {
 		log.Fatalf("Invalid apiVersion and kind: (%v)", err)
 	}
 
-	s := &scaffold.Scaffold{}
-	tmpdir, err := ioutil.TempDir("", "osdk")
-	if err != nil {
-		log.Fatalf("Unable to get temp directory: (%v)", err)
+	roleFiles := ansible.RolesFiles{
+		Resource: *resource,
 	}
-
-	galaxyInit := &ansible.GalaxyInit{
+	roleTemplates := ansible.RolesTemplates{
 		Resource: *resource,
-		Dir:      tmpdir,
 	}
 
+	s := &scaffold.Scaffold{}
 	err = s.Execute(cfg,
-		&ansible.Dockerfile{
-			GeneratePlaybook: generatePlaybook,
-		},
-		&ansible.WatchesYAML{
-			Resource:         *resource,
-			GeneratePlaybook: generatePlaybook,
-		},
-		galaxyInit,
 		&scaffold.ServiceAccount{},
 		&scaffold.Role{
 			IsClusterScoped: isClusterScoped,
 		},
 		&scaffold.RoleBinding{
 			IsClusterScoped: isClusterScoped,
 		},
-		&ansible.Operator{
-			IsClusterScoped: isClusterScoped,
-		},
 		&scaffold.Crd{
 			Resource: resource,
 		},
 		&scaffold.Cr{
 			Resource: resource,
 		},
+		&ansible.BuildDockerfile{
+			GeneratePlaybook: generatePlaybook,
+		},
+
+		&ansible.RolesReadme{
+			Resource: *resource,
+		},
+
+		&ansible.RolesMetaMain{
+			Resource: *resource,
+		},
+		&roleFiles,
+		&roleTemplates,
+
+		&ansible.RolesVarsMain{
+			Resource: *resource,
+		},
+
+		&ansible.MoleculeTestLocalPlaybook{
+			Resource: *resource,
+		},
+
+		&ansible.RolesDefaultsMain{
+			Resource: *resource,
+		},
+
+		&ansible.RolesTasksMain{
+			Resource: *resource,
+		},
+
+		&ansible.MoleculeDefaultMolecule{},
+
+		&ansible.BuildTestFrameworkDockerfile{},
+
+		&ansible.MoleculeTestClusterMolecule{},
+
+		&ansible.MoleculeDefaultPrepare{},
+
+		&ansible.MoleculeDefaultPlaybook{
+			GeneratePlaybook: generatePlaybook,
+			Resource:         *resource,
+		},
+
+		&ansible.BuildTestFrameworkAnsibleTestScript{},
+
+		&ansible.MoleculeDefaultAsserts{},
+
+		&ansible.MoleculeTestClusterPlaybook{
+			Resource: *resource,
+		},
+
+		&ansible.RolesHandlersMain{
+			Resource: *resource,
+		},
+
+		&ansible.Watches{
+			GeneratePlaybook: generatePlaybook,
+			Resource:         *resource,
+		},
+
+		&ansible.DeployOperator{
+			IsClusterScoped: isClusterScoped,
+		},
+
+		&ansible.Travis{},
+
+		&ansible.MoleculeTestLocalMolecule{},
+
+		&ansible.MoleculeTestLocalPrepare{
+			Resource: *resource,
+		},
 	)
 	if err != nil {
 		log.Fatalf("New ansible scaffold failed: (%v)", err)
 	}
 
+	// Remove placeholders from empty directories
```

Comment:
Would these be a good fit for a `defer` that happens back when the files get created?